Find a functionally equivalent code:android.transition.TransitionSet.runAnimators:COMMENT
Method Modifier: protected   hidden      
Comment:/**
 * @hide
 */

@@ -1,8 +1,14 @@
 {
+    if (mTransitions.isEmpty()) {
+        start();
+        end();
+        return;
+    }
     setupStartEndListeners();
+    int numTransitions = mTransitions.size();
     if (!mPlayTogether) {
         // TODO: Need to add listeners in such a way that we can remove them later if canceled
-        for (int i = 1; i < mTransitions.size(); ++i) {
+        for (int i = 1; i < numTransitions; ++i) {
             Transition previousTransition = mTransitions.get(i - 1);
             final Transition nextTransition = mTransitions.get(i);
             previousTransition.addListener(new TransitionListenerAdapter() {
@@ -19,26 +25,32 @@
             firstTransition.runAnimators();
         }
     } else {
-        for (Transition childTransition : mTransitions) {
-            childTransition.runAnimators();
+        for (int i = 0; i < numTransitions; ++i) {
+            mTransitions.get(i).runAnimators();
         }
     }
 }

